src/main/java/com/rultor/agents/ec2/AmazonEC2.java

```diff
@@ -0,0 +1,146 @@
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.InstanceState;
+import com.amazonaws.services.ec2.model.RunInstancesRequest;
+import com.jcabi.aspects.Tv;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Amazon EC2 instance on demand. Takes all the mandatory data for running
+ * an EC2 instance in a defined region when it's needed.
+ *
+ * @author Marton Horvath (marton.horvath@m323.org)
+ * @version $Id$
+ * @since 2.0
+ */
+public class AmazonEC2 implements Amazon {
+    /**
+     * The integer value of the Running {@link InstanceState}.
+     */
+    private static final int RUNNING_STATE = 16;
+    /**
+     * AWS credentials used when authenticating at AWS.
+     */
+    private final transient AWSCredentials credentials;
+    /**
+     * AWS region to be used for the instance to be run.
+     */
+    private final transient String region;
+    /**
+     * EC2 instance type (like <i>t1.micro</i>, <i>m1.small</i> etc.).
+     */
+    private final transient String type;
+    /**
+     * EC2 instance key pair name.
+     */
+    private final transient String keyname;
+
+    /**
+     * Public ctor.
+     * @param cred Credentials for AWS
+     * @param zone AWS region
+     * @param instype Instance type
+     * @param key Instance key pair name
+     * @checkstyle ParameterNumberCheck (12 lines)
+     */
+    public AmazonEC2(
+        @NotNull final AWSCredentials cred,
+        @NotNull final String zone,
+        @NotNull final String instype,
+        @NotNull final String key) {
+        super();
+        this.credentials = cred;
+        this.region = zone;
+        this.type = instype;
+        this.keyname = key;
+    }
+
+    @Override
+    public final Instance runOnDemand() throws IOException {
+        Instance instance;
+        final AmazonEC2Client client = new AmazonEC2Client(
+            this.credentials
+        );
+        client.setEndpoint(this.region);
+        final List<Instance> instances = client.runInstances(
+            this.create()
+        ).getReservation().getInstances();
+        if (instances.isEmpty()) {
+            throw new IllegalStateException(
+                "No Instance was available in the RunInstanceRequest reply!"
+            );
+        } else {
+            instance = instances.get(0);
+        }
+        InstanceState state = instance.getState();
+        if (state.getCode() < RUNNING_STATE) {
+            final DescribeInstanceStatusRequest statusreq =
+                new DescribeInstanceStatusRequest();
+            statusreq.withInstanceIds(instance.getInstanceId());
+            while (state.getCode() < RUNNING_STATE) {
+                try {
+                    TimeUnit.SECONDS.sleep(Tv.FIFTEEN);
+                } catch (final InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                    break;
```

Comment:
why swallowing the exception? why can't we just throw it up, wrapped into `IllegalStateException`